iot-hub/Samples/service/ImportExportDevicesWithManagedIdentitySample/ImportExportDevicesWithManagedidentitySample.cs

```diff
@@ -48,7 +48,7 @@ public class ImportExportDevicesWithManagedidentitySample
             // the job will fail.
             // If StorageAuthenticationType is set to IdentityBased the userAssignedIdentity property is
             // null, the jobs will use system defined identity. If the IoT hub is not configured with the
-            // user defined managed identity, the job will fail.
+            // system defined managed identity, the job will fail.
```

Comment:
This is incorrect. The "user defined managed identity" should be configured if StorageAuthenticationType is set to IdentityBased. I think I get why you made this change but the intent was to tell this has to be configured.

Maybe we need two separate sentences -
1) If StorageAuthenticationType is set to IdentityBased but IoT hub is not configured with the user identity, the job will fail.
2) If StorageAuthenticationType is set to IdentityBased the userAssignedIdentity property is set to null in the payload, it will default to using system defined identity automatically.

Reply:
I think in (1) you mean "Identity" and not "user identity", right?

Reply:
For the ImportExportDevicesWithManagedidentitySample, there are 3 options:

StorageAuthenticationType is set to IdentityBased and neither user defined nor system defined managed identities are configured on the hub => job fails
StorageAuthenticationType is set to IdentityBased and user defined identity is configured in hub => user identity passed in matches the one configured in hub => pass, else fail
StorageAuthenticationType is set to IdentityBased and user defined identity passed is null => system defined identity needs to be configured in hub => configured then pass, else fail

These 3 scenarios are covered here:

```
// If StorageAuthenticationType is set to IdentityBased and userAssignedIdentity property is
// not null, the jobs will use user defined managed identity. If the IoT hub is not
// configured with the user defined managed identity specified in userAssignedIdentity,
// the job will fail.
// If StorageAuthenticationType is set to IdentityBased and userAssignedIdentity property is
// null, the jobs will use system defined identity. If the IoT hub is not configured with the
// system defined managed identity, the job will fail.
// If StorageAuthenticationType is set to IdentityBased and neither user defined nor system defined
// managed identities are configured on the hub, the job will fail.
```

So I feel the statement is correct. Let me know if I am missing something.

Reply:
The intent of this sample is just to test user-defined. But bcoz of the way the service has implemented it, you can still end up with a success and think you used user defined. My intent here was to point that out.
// If StorageAuthenticationType is set to IdentityBased the userAssignedIdentity property is
// null, the jobs will use system defined identity by default. If the IoT hub is configured with the
// system defined managed identity, the job will succeed but will not use the user defined managed identity. 
// Othwersie, the job will fail.

How does this sound?